### ISSUE TITLE
deno test doesn't work on subdirs

### ISSUE THREAD

**Issue by ry** [denoland/deno_std#569]
```
~/src/deno_std> deno test ./fmt/
Found 4 matching test files.
error: Uncaught SyntaxError: Unexpected token '*'
► file:///Users/rld/src/deno_std/fmt/TODO:2:1

2 * "native" formatting, json, arrays, object/structs, functions ...
  ^

```

**Comment by bartlomieju** [comment]
I don't think it's a bug. `./fmt` is converted to a regex to match files during `walk()` and it causes to match all files in `fmt/` directory. 

Actually the error is the same as in #568 - importing non-JS file. For good measure we can always only match `.js` and `.ts` files, but it may be surprising that some files are omitted. Anyway... `deno test` should definitely be as ergonomic as possible, we should also check how other test runners handle that.

**Comment by bartlomieju** [comment]
Nevermind my previous comment. This issue will be fixed in #574 as well. Test runner will import all files from that directory obviously, and only some of them contain tests :+1:
